include/fx/gltf.h

```diff
@@ -27,6 +27,28 @@
 #define FX_GLTF_NODISCARD
 #endif
 
+#if defined(__clang__)
+    #if __clang_major__ < 7 || (defined(__cplusplus) && __cplusplus < 201703L)
+        #define  FX_GLTF_EXPERIMENTAL_FILESYSTEM
+    #endif
+#elif defined(__GNUC__)
+    #if __GNUC__ < 8 || (defined(__cplusplus) && __cplusplus < 201703L)
+        #define  FX_GLTF_EXPERIMENTAL_FILESYSTEM
+    #endif
+#elif defined(_MSC_VER)
+    #if _MSC_VER < 1914 || (!defined(_HAS_CXX17) || (defined(_HAS_CXX17) && _HAS_CXX17 == 0))
+        #define  FX_GLTF_EXPERIMENTAL_FILESYSTEM
+    #endif
+#endif
+
+#ifdef FX_GLTF_EXPERIMENTAL_FILESYSTEM
+    #include <experimental/filesystem>
+    #define FX_GLTF_FILESYSTEM std::experimental::filesystem::v1
+#else
+    #include <filesystem>
+    #define FX_GLTF_FILESYSTEM std::filesystem

```

Comment:
Can you undefine both of these defines at the very bottom of this file.  I try not to let any defines escape outside the header file so folks don't start depending on them.

I know that means they might have to do a similar dance in order to load the right filesystem header but I think that's actually a good thing for them to do on their side.

Reply:
Thanks for your suggestion @jessey-git! I didn't see the undef at the end of the file. I've modified my commit to harmonize the code.